Is xformers still not support cuda 12.0?

# ❓ Questions and Help
I got the following error while installing...


Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com/
Obtaining file:///F:/Stable_Diffusion/stable-diffusion-webui-master/repositories/xformers
Preparing metadata (setup.py) ... error
error: subprocess-exited-with-error

× python setup.py egg_info did not run successfully.
│ exit code: 1
╰─> [9 lines of output]
No CUDA runtime is found, using CUDA_HOME='C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.0'
Traceback (most recent call last):
File "", line 2, in
File "", line 34, in
File "F:\Stable_Diffusion\stable-diffusion-webui-master\repositories\xformers\setup.py", line 293, in
symlink_package(
File "F:\Stable_Diffusion\stable-diffusion-webui-master\repositories\xformers\setup.py", line 83, in symlink_package
os.symlink(src=path_from, dst=path_to)
OSError: [WinError 1314] A required privilege is not held by the client: 'F:\Stable_Diffusion\stable-diffusion-webui-master\repositories\xformers\third_party\flash-attention\flash_attn' -> 'F:\Stable_Diffusion\stable-diffusion-webui-master\repositories\xformers\xformers\_flash_attn'
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

Is it for cuda 12. Should I degrade the cuda version?

or what is the problem, can anyone help?

Hello,

I tried with the patch but got an error again. I already changed my cuda version 11.7. 
This is the cmd output (It's very big). 
the last part error is 

 C:\Users\User\AppData\Local\Temp\tmpxft_00001a64_00000000-7_fmha_block_dgrad_fp16_kernel_loop.sm80.compute_86.cudafe1.cpp : fatal error C1083: Cannot open compiler generated file: '': Invalid argument
    error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v11.7\\bin\\nvcc.exe' failed with exit code 4294967295

--- a/setup.py
+++ b/setup.py
@@ -76,11 +76,21 @@ def symlink_package(name: str, path: Path) -> None:
     cwd = Path(__file__).parent
     path_from = cwd / path
     path_to = os.path.join(cwd, *name.split("."))
+
+    # OSError: [WinError 1314] A required privilege is not held by the client
+    # Windows requires special permission to symlink. Fallback to copy
+    use_symlink = os.name != "nt"
     try:
-        os.remove(path_to)
+        if use_symlink:
+            os.remove(path_to)
+        else:
+            shutil.rmtree(path_to)
     except FileNotFoundError:
         pass
-    os.symlink(src=path_from, dst=path_to)
+    if use_symlink:
+        os.symlink(src=path_from, dst=path_to)
+    else:
+        shutil.copytree(src=path_from, dst=path_to)
 
 
 def get_cuda_version(cuda_dir) -> int:

